extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1031,6 +1049,52 @@ export class Project implements ISqlProject {
 			}
 		}
 	}
+
+	/**
+	 * Evaluates the value of the property item in the loaded project.
+	 *
+	 * @param propertyName Name of the property item to evaluate.
+	 * @returns Value of the property or `undefined`, if property is missing.
+	 */
+	private evaluateProjectPropertyValue(propertyName: string): string | undefined;
```

Comment:
why are these needed? Looks like defaultValue is already an optional parameter on line 1077

Reply:
This is how overloads are defined in TypeScript. The return type is different, depending on whether or not the default value provided. This eliminates the need of additional null check when default value is specified.